src/test/java/formflow/library/utilities/AbstractMockMvcTest.java

```diff
@@ -395,7 +404,14 @@ protected void assertInputHasErrors(String pageName, String inputName, Integer n
   protected void assertInputHasErrors(String pageName, String inputName, List<String> errorMessages)
       throws Exception {
     var page = new FormScreen(getPage(pageName));
-    assertTrue(errorMessages.containsAll(page.getInputErrors(inputName).stream().map(Element::ownText).toList()));
+    // make sure there are errors returned
+    assertFalse(page.getInputErrors(inputName).isEmpty(), "Expected errors on page, but there were none");
+    // assert equal amount
+    assertEquals(errorMessages.size(), page.getInputErrors(inputName).size(),
+        "The page has a different number of errors than expected");
+    // now check if they match the expected ones.
+    assertTrue(errorMessages.containsAll(page.getInputErrors(inputName).stream().map(Element::ownText).toList()),
+        "The error messages expected do match what was returned");
   }
```

Comment:
This function needed more. There were two issues with the current check: 
 * The logic of `List.containsAll(empty List)` will return true, since a set always contains an empty set.  So the logic here was that if the page hadn't POSTed and there were no errors, this function thought things were fine, since no errors is a subset of the errors list. 
 * Also this didn't take into account if the number of errors returned by the page were less than the ones expected. It just makes sure that the error messages list contains all that the page returned.   If the page returned less errors than expected, it still thought things were fine. 

Adding the other two asserts should account for these scenarios.